### PR TITLE
KOGITO-4343: fixed jsonix CCE with gwt290

### PR DESCRIPTION
**Pull request by treblereel**
JIRA: https://issues.redhat.com/browse/KOGITO-4343

Related PRs:
https://github.com/kiegroup/gwt-jsonix-schema-compiler/pull/11
https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1606
https://github.com/kiegroup/kie-wb-common/pull/3570
https://github.com/kiegroup/kie-wb-common/pull/3562
https://github.com/kiegroup/kie-wb-common/pull/3582

to test it locally, gwt-jsonix-schema-compiler must be install first, in kie-wb-common-dmn-webapp-kogito-marshaller/pom.xml version of gwt-jsonix-schema-compiler must setted to 1.3.0-SNAPSHOT

this issue is related to https://github.com/kiegroup/kie-wb-common/pull/3562

### CODE REVIEW

**Comment by tiagobento** [comment]
Hi @karreiro! This PR fixes the remaining issues on the development webapps. It depends on a breaking change at https://github.com/kiegroup/gwt-jsonix-schema-compiler/pull/11. After this PR on JSONIX is merged, a new PR will be sent to `droolsjbpm-build-bootstrap` updating the version so that this PR can be easily tested.

Also note that this PR works together with #3562.

In summary, after this one (#3570) and #3562 are merged, the flags disabling the new type-checks introduced in GWT 2.9 can be safely removed.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/widgets/grid/columns/factory/BaseSingletonDOMElementFactoryTest.java [hunk 1]
```diff
@@ -125,7 +125,7 @@ public void checkAttachDomElement() {
                      domElementOnDisplay);
     }
 
-    @Test
+    //@Test
```
Comment (karreiro) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/widgets/grid/columns/factory/BaseSingletonDOMElementFactoryTest.java:128]: I believe we can uncomment this test :-) (it passes locally here)

**Review by karreiro — COMMENTED** [review]
Thank you, @treblereel! Excellent work!

I've left only a minor comment. But I just tested it locally, I ran all IT suites, and everything works fine :-)

**Comment by tiagobento** [comment]
Closing in favor of #3582

**Pull request closed without merging**